src/txn/txn.c

```diff
@@ -1213,10 +1217,14 @@ __wt_txn_rollback(WT_SESSION_IMPL *session, const char *cfg[])
 			 * Need to resolve indirect references of transaction
 			 * operation, in case of prepared transaction.
 			 */
-			if (F_ISSET(txn, WT_TXN_PREPARE))
-				WT_RET(__wt_txn_resolve_prepared_op(
-				    session, op, false));
-			else {
+			if (F_ISSET(txn, WT_TXN_PREPARE)) {
+#ifdef HAVE_DIAGNOSTIC
+				if (!F_ISSET(op, WT_TXN_MOD_REPEATED))
+#endif
+					WT_RET(__wt_txn_resolve_prepared_op(
+					    session, op, false,
+					    &resolved_update_count));
```

Comment:
This would be a good place to send in NULL and let the `wt_txn_resolve_prepared_op` function handle accepting a NULL pointer. It would mostly remove all the changes that are not really needed in this function.